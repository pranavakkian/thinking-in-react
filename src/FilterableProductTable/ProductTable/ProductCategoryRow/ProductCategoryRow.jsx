import PropTypes from 'prop-types';
import React, { memo } from 'react';

const ProductCategoryRow = (props) => {
  const { category } = props;
  return (
    <th id="categoryHeader" colSpan="2">
      {category}
    </th>
  );
};

ProductCategoryRow.propTypes = {
  category: PropTypes.string.isRequired,
};

export default memo(ProductCategoryRow);
