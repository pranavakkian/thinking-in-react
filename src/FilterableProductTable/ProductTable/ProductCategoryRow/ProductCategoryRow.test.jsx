/* eslint-disable no-undef */
import { mount } from 'enzyme';
import React from 'react';
import ProductCategoryRow from './ProductCategoryRow';

const wrapper = mount(<ProductCategoryRow category="Hello" />);

describe('ProductCategoryRow', () => {
  it('ProductCategoryRow rendered successfully!', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Table header rendered correctly!', () => {
    const tableHeader = wrapper.find('th');
    expect(tableHeader).toHaveLength(1);
    expect(tableHeader.text()).toEqual('Hello');
    expect(tableHeader.props().colSpan).toEqual('2');
  });

  it('ProductCategoryRow un-mounted successfully!', () => {
    wrapper.unmount();
  });
});
