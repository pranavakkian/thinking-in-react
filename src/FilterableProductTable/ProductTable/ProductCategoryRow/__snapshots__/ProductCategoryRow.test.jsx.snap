// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCategoryRow ProductCategoryRow rendered successfully! 1`] = `
<Memo(ProductCategoryRow)
  category="Hello"
>
  <th
    colSpan="2"
    id="categoryHeader"
  >
    Hello
  </th>
</Memo(ProductCategoryRow)>
`;
